WikiSentiment/WikiAPIRequests.cs

```diff
@@ -0,0 +1,164 @@
+            if (articleText.StartsWith("#") &&
+                articleText.Contains("[[") &&
+                articleText.Contains("]]"))
+            {
+                return normalizeTitle(articleText.Split("[[", 2)[1].Split("]]")[0]);
+            }
+
+            return "";
+        }
+
+        /// <summary>
+        /// Get top pairs of (title, views) for given language
+        /// </summary>
+        /// <param name="_client">Http client with API headers</param>
+        /// <param name="_languageCode">Two letter country code</param>
+        /// <param name="_date">Date for lookup</param>
+        /// <returns>Tuple with articleArray(name, views) of articles</returns>
+        public static async Task<(string title, int views)[]>
+            GetArticleList(HttpClient _client, string _languageCode, DateTime _date)
+        {
+            var articlesString = await _client.GetStringAsync(
+                    "https://wikimedia.org/api/rest_v1/metrics/pageviews/" +
+                    $"top/{_languageCode}.wikipedia.org/all-access/{_date.Year}/{_date.Month:D2}/{_date.Day:D2}");
+
+            var jArticleArray = JsonNode.Parse(articlesString).AsObject()["items"][0]["articles"].AsArray();
+
+            var articleArray = new (string title, int views)[jArticleArray.Count];
+
+            for (int i = 0; i < jArticleArray.Count; i++)
+            {
+                articleArray[i] = (
+                    normalizeTitle((string)jArticleArray[i]["article"].AsValue()), 
+                    (int)jArticleArray[i]["views"].AsValue());
+            }
+
+            return articleArray;
+        }
+
+        /// <summary>
+        /// Gets total daily views for given language
+        /// </summary>
+        /// <param name="_client"></param>
+        /// <param name="_languageCode"></param>
+        /// <param name="_date"></param>
+        /// <returns></returns>
+        public static async Task<int> GetTotalViews(
+            HttpClient _client, string _languageCode, DateTime _date)
+        {
+            var viewsString = await _client.GetStringAsync(
+                    "https://wikimedia.org/api/rest_v1/metrics/pageviews/" +
+                    $"aggregate/{_languageCode}.wikipedia.org/all-access/user/daily/" +
+                    $"{_date.Year}{_date.Month:D2}{_date.Day:D2}/{_date.Year}{_date.Month:D2}{_date.Day:D2}");
+
+            JsonObject jObject = JsonNode.Parse(viewsString).AsObject();
+
+            //validate object and return total views
+            if (jObject.ContainsKey("items") &&
+                jObject["items"].AsArray().Count > 0 &&
+                jObject["items"][0].AsObject().ContainsKey("views") )
+            {
+                return (int)JsonNode.Parse(viewsString).AsObject()["items"][0]["views"];
+            }
+
+            throw new HttpRequestException($"Http response has bad format. Date: {_date}; language:{_languageCode}");
```

Comment:
Don't make the exceptional case the default.

```suggestion
            //validate object and return total views
            if (!(jObject.ContainsKey("items") &&
                  jObject["items"].AsArray().Count > 0 &&
                  jObject["items"][0].AsObject().ContainsKey("views") ))
            {
                            throw new HttpRequestException($"Http response has bad format. Date: {_date}; language:{_languageCode}");
            }
            return (int)JsonNode.Parse(viewsString).AsObject()["items"][0]["views"];


```